go/services/reports/pkg/scheduled_jobs/scheduled_job_repository.go

```diff
@@ -0,0 +1,175 @@
+package scheduledjobs
+
+import (
+	"context"
+	"fmt"
+	"github.com/IBM/fp-go/array"
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/repositories"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.uber.org/zap"
+)
+
+type ScheduledJobRepositoryErrorKind string
+
+type ScheduledJob interface {
+	GetId() string
+	SetId(newId string)
+	GetStatus() string
+}
+
+const (
+	ScheduledJobNotFound      ScheduledJobRepositoryErrorKind = "SCHEDULED_JOB_NOT_FOUND"
+	InvalidScheduledJobId     ScheduledJobRepositoryErrorKind = "INVALID_SCHEDULED_JOB_ID"
+	ScheduledJobInternalError ScheduledJobRepositoryErrorKind = "INTERNAL_ERROR"
+)
+
+type ScheduledJobRepositoryError struct {
+	msg  string
+	kind ScheduledJobRepositoryErrorKind
+}
+
+func (e *ScheduledJobRepositoryError) Error() string {
+	return e.msg
+}
+
+func (e *ScheduledJobRepositoryError) Kind() ScheduledJobRepositoryErrorKind {
+	return e.kind
+}
+
+func NewScheduledJobNotFoundError(err error) *ScheduledJobRepositoryError {
+	return &ScheduledJobRepositoryError{
+		msg:  fmt.Sprintf(" incident does not exists: %s", err),
+		kind: ScheduledJobNotFound,
+	}
+}
+
+func NewInvalidScheduledJobIdError(err error) *ScheduledJobRepositoryError {
+	return &ScheduledJobRepositoryError{
+		msg:  fmt.Sprintf("Invalid application incident id: %s", err),
+		kind: InvalidScheduledJobId,
+	}
+}
+
+func NewScheduledJobInternalError(err error) *ScheduledJobRepositoryError {
+	return &ScheduledJobRepositoryError{
+		msg:  err.Error(),
+		kind: ScheduledJobInternalError,
+	}
+}
+
+type ScheduledJobRepository[T ScheduledJob] interface {
+	InsertScheduledJobs(ctx context.Context, jobs []T) ([]string, *ScheduledJobRepositoryError)
+	GetScheduledJob(ctx context.Context, id string) (*T, *ScheduledJobRepositoryError)
+	GetScheduledJobsByIds(ctx context.Context, ids []string) ([]T, *ScheduledJobRepositoryError)
+	GetScheduledJobsByStatus(ctx context.Context, status string) ([]T, *ScheduledJobRepositoryError)
+	UpdateScheduledJob(ctx context.Context, job T) *ScheduledJobRepositoryError
+}
+
+type MongoDbScheduledJobRepository[T ScheduledJob] struct {
+	mongoDbCollection *repositories.MongoDbCollection[T]
+	logger            *zap.Logger
+}
+
+func NewMongoDbScheduledJobRepository[T ScheduledJob](mongoDbCollection *repositories.MongoDbCollection[T], logger *zap.Logger) *MongoDbScheduledJobRepository[T] {
+	return &MongoDbScheduledJobRepository[T]{
+		mongoDbCollection: mongoDbCollection,
+		logger:            logger,
+	}
+}
+
+func (r *MongoDbScheduledJobRepository[T]) InsertScheduledJobs(ctx context.Context, incidents []T) ([]string, *ScheduledJobRepositoryError) {
+
+	documents := make([]interface{}, 0, len(incidents))
+
+	if len(incidents) == 0 {
```

Comment:
I wonder if this check is needed, will len(incidents) == 0 cause inconvenience in further operations?

Reply:
This was added because of the mongodb driver. It returns an error when it gets an empty slice